lib/private/Files/FileInfo.php

```diff
@@ -181,7 +181,11 @@ public function getMimePart() {
 	 * @return string
 	 */
 	public function getName() {
-		return isset($this->data['name']) ? $this->data['name'] : basename($this->getPath());
+		if (isset($this->data['name']) && !empty($this->data['name'])) {
+			return $this->data['name'];
+		}
+
+		return basename($this->getPath());
```

Comment:
`empty` should be enough:
```suggestion
		return empty($this->data['name'])
			? basename($this->getPath())
			: $this->data['name'];
```

Reply:
> `empty` should be enough:

I agree, isset is redundant. I corrected the code